app/server/appsmith-server/src/main/java/com/appsmith/server/services/MockDataServiceImpl.java

```diff
@@ -174,10 +174,17 @@ private DatasourceConfiguration getPostgresDataSourceConfiguration(String name,
     private Mono<Datasource> createSuffixedDatasource(Datasource datasource, String name, int suffix) {
         final String actualName = name + (suffix == 0 ? "" : " (" + suffix + ")");
         datasource.setName(actualName);
+        String password = null;
+        if( datasource.getDatasourceConfiguration().getAuthentication() instanceof DBAuth) {
+            password = ((DBAuth) datasource.getDatasourceConfiguration().getAuthentication()).getPassword();
+        }
+        final String finalPassword = password;
         return datasourceService.create(datasource)
                 .onErrorResume(DuplicateKeyException.class, error -> {
                     if (error.getMessage() != null
-                            && error.getMessage().contains("organization_datasource_deleted_compound_index")) {
+                            && error.getMessage().contains("organization_datasource_deleted_compound_index")
+                            && datasource.getDatasourceConfiguration().getAuthentication() instanceof DBAuth) {
+                        ((DBAuth) datasource.getDatasourceConfiguration().getAuthentication()).setPassword(finalPassword);
                         return createSuffixedDatasource(datasource, name, 1 + suffix);
```

Comment:
If I understand correctly, if the auth for the datasource is _not_ `DBAuth`, then we don't create a suffixed datasource at all? Do we not want to create a suffixed datasource even if the auth isn't a `DBAuth`? Something like:

```suggestion
                            && error.getMessage().contains("organization_datasource_deleted_compound_index") {
                        if (datasource.getDatasourceConfiguration().getAuthentication() instanceof DBAuth) {
                            ((DBAuth) datasource.getDatasourceConfiguration().getAuthentication()).setPassword(finalPassword);
                        }
                        return createSuffixedDatasource(datasource, name, 1 + suffix);
```

Reply:
No, we dont want create a datasource if the type is not `dbAuth`. Since the mock datasource is of db type. The current implementation supports only the DB type datasource creation.

Reply:
I see. So, if we need mock datasources that are not of DB type, we'll need code changes then. Alright, approving.